### ISSUE TITLE
Popping a stash errors with (wrong-type-argument listp "stash@{0}")

### ISSUE THREAD

**Issue by tsdh** [magit/magit#4971]
I have a stash and hit `z p` in the magit status buffer.  I am queried if I want to pop it, confirm with yes, and then get this error:

```
Debugger entered--Lisp error: (wrong-type-argument listp "stash@{0}")
  apply(magit-git-items "stash" "show" "-z" "--name-only" "stash@{0}")
  magit-stashed-files("stash@{0}")
  #f(compiled-function (stash) "Apply a stash to the working tree and remove it from stash list.\nIf nothing is staged, then try to reinstate the stashed index.\nDoing so is not possible if there are staged changes.  Do not\nremove the stash, if it cannot be applied." (interactive #f(compiled-function () #<bytecode 0x142bc511d145fcdd>)) #<bytecode 0x13e21ab4a37b510f>)("stash@{0}")
  apply(#f(compiled-function (stash) "Apply a stash to the working tree and remove it from stash list.\nIf nothing is staged, then try to reinstate the stashed index.\nDoing so is not possible if there are staged changes.  Do not\nremove the stash, if it cannot be applied." (interactive #f(compiled-function () #<bytecode 0x142bc511d145fcdd>)) #<bytecode 0x13e21ab4a37b510f>) "stash@{0}")
  #f(compiled-function (fn &rest args) #<bytecode -0x1044cb1198c2d1f3>)(#f(compiled-function (stash) "Apply a stash to the working tree and remove it from stash list.\nIf nothing is staged, then try to reinstate the stashed index.\nDoing so is not possible if there are staged changes.  Do not\nremove the stash, if it cannot be applied." (interactive #f(compiled-function () #<bytecode 0x142bc511d145fcdd>)) #<bytecode 0x13e21ab4a37b510f>) "stash@{0}")
  apply(#f(compiled-function (fn &rest args) #<bytecode -0x1044cb1198c2d1f3>) #f(compiled-function (stash) "Apply a stash to the working tree and remove it from stash list.\nIf nothing is staged, then try to reinstate the stashed index.\nDoing so is not possible if there are staged changes.  Do not\nremove the stash, if it cannot be applied." (interactive #f(compiled-function () #<bytecode 0x142bc511d145fcdd>)) #<bytecode 0x13e21ab4a37b510f>) "stash@{0}")
  (lambda (fn &rest args) (interactive #f(compiled-function (spec) #<bytecode 0x12244f3d3770ad12>)) (apply '#f(compiled-function (fn &rest args) #<bytecode -0x1044cb1198c2d1f3>) fn args))(#f(compiled-function (stash) "Apply a stash to the working tree and remove it from stash list.\nIf nothing is staged, then try to reinstate the stashed index.\nDoing so is not possible if there are staged changes.  Do not\nremove the stash, if it cannot be applied." (interactive #f(compiled-function () #<bytecode 0x142bc511d145fcdd>)) #<bytecode 0x13e21ab4a37b510f>) "stash@{0}")
  apply((lambda (fn &rest args) (interactive #f(compiled-function (spec) #<bytecode 0x12244f3d3770ad12>)) (apply '#f(compiled-function (fn &rest args) #<bytecode -0x1044cb1198c2d1f3>) fn args)) #f(compiled-function (stash) "Apply a stash to the working tree and remove it from stash list.\nIf nothing is staged, then try to reinstate the stashed index.\nDoing so is not possible if there are staged changes.  Do not\nremove the stash, if it cannot be applied." (interactive #f(compiled-function () #<bytecode 0x142bc511d145fcdd>)) #<bytecode 0x13e21ab4a37b510f>) "stash@{0}")
  magit-stash-pop("stash@{0}")
  funcall-interactively(magit-stash-pop "stash@{0}")
  call-interactively(magit-stash-pop nil nil)
  command-execute(magit-stash-pop)
```

The problem is
```elisp
(defun magit-stashed-files (stash)
  (apply #'magit-git-items "stash" "show" "-z" "--name-only" stash))
```
where the last arg `stash` is a string and not a list.

    M-x magit-version RET

Magit 20230723.1601 [>= 3.3.0.50-git], Transient 0.4.1, Git 2.41.0, Emacs 30.0.50, gnu/linux

**Comment by tsdh** [comment]
This will be fixed by #4970 which I have just discovered.

**Comment by kyleam** [comment]
Thanks for reporting @tsdh.  gh-4970 has landed.
